### DISCUSSION TITLE
GitHub Actions Expressions would benefit from jq-like syntax

### DISCUSSION THREAD

**Discussion by giner** [community/community#37588] [Actions]
The [Expressions](https://docs.github.com/en/actions/learn-github-actions/expressions) are currently limited to a small set of operations. This results in people writing actions to do such a simple thing as splitting strings:
- https://github.com/marketplace/actions/split-action
- https://github.com/marketplace/actions/split-by
- https://github.com/marketplace/actions/simple-split
- https://github.com/orgs/community/discussions/26625
- https://github.com/orgs/community/discussions/25768

Which does not only make workflows heavy but also makes it impossible to use these expressions at the pre-processing step before any job runs. How about adding flexible and time-proven [jq-like](https://stedolan.github.io/jq/tutorial/) language to the Expressions?

**Comment by ossanna16** [comment]
Hi there @giner and welcome to our community! Thank you for asking a great question 🙂
While you are waiting for a reply to your question, here are a few things you can do:

To get started, introduce yourself in our [official introduction thread](https://github.com/orgs/community/discussions/23204)
To connect with other new users, check out this Discussion to [Meet Your GitHub Community Team](https://github.com/orgs/community/discussions/28776), further your GitHub knowledge by working through our [GitHub Skill 🆙 exercises](https://github.com/orgs/community/discussions/32121), and participate in our [weekly Fun Friday chats](https://github.com/orgs/community/discussions/30912)
